.github/workflows/ci-cd.yml

```
@@ -23,10 +23,25 @@ env:
   FORCE_COLOR: 1  # Request colored output from CLI tools supporting it
   MYPY_FORCE_COLOR: 1
   PY_COLORS: 1
+  UPSTREAM_REPOSITORY_ID: >-
+    13258039
 
 permissions: {}
+
 jobs:
 
+  pre-setup:
+    name: Pre-Setup global build settings
+    runs-on: ubuntu-latest
+    outputs:
+      upstream-repository-id: ${{ env.UPSTREAM_REPOSITORY_ID }}
+      release-requested: >-
+        ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags/') }}
```

Nowadays, this check can be a bit more elegant. Also let's keep multiline. Additionally, ternary is needed as otherwise GH might render this into something weird like an empty string IIRC. So this is my standard idiom across projects.

```suggestion
        ${{
          (
            github.event_name == 'push'
            && github.ref_type == 'tag'
          )
          && true
          || false
        }}
```